### PR TITLE
Updating orderForm in cart and checkout when necessary

### PR DESCRIPTION
**Pull request by silviasfon**
#### What problem is this solving?

This pull request guarantees that the orderForm will always be up to date when the consumer gets to the cart/checkout pages.

#### How should this be manually tested?

[Here](https://refresh--checkoutio.myvtex.com/) add any item to the cart (don't close the page). Then, run [this](https://refresh--checkoutio.myvtex.com/_v/private/vtex.order-manager@0.9.0/graphiql/v1?query=mutation%20%7B%0A%20%20updateItems(orderItems%3A%20%7B%0A%20%20%20%20index%3A%200%2C%0A%20%20%20%20quantity%3A%203%0A%20%20%7D%2C%20allowedOutdatedData%3A%20%5B%22paymentData%22%5D)%20%7B%0A%20%20%20%20value%0A%20%20%20%20paymentData%7B%0A%20%20%20%20%20%20installmentOptions%7B%0A%20%20%20%20%20%20%20%20value%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D) mutation and check that the field `value` in `updateItems` differs from the `value` fields in `installmentOptions`. Then, on the first page, go to the cart or the checkout page. Finally, run [this](https://refresh--checkoutio.myvtex.com/_v/private/vtex.order-manager@0.9.0/graphiql/v1?query=query%20%7B%0A%20%20orderForm%7B%0A%20%20%20%20value%0A%20%20%20%20paymentData%7B%0A%20%20%20%20%20%20installmentOptions%7B%0A%20%20%20%20%20%20%20%20value%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D) query and check that the fields that were different are now equal. 

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Jira story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: react/OrderForm.tsx [hunk 1]
```diff
@@ -77,11 +80,17 @@ const getLocalOrderForm = (): OrderForm | null => {
 
 export const OrderFormProvider: FC = ({ children }) => {
   const { logSplunk } = useSplunk()
+  const runtime = useRuntime
+  const page = runtime().page
```
Comment (lucasecdb) [inline react/OrderForm.tsx:84]: this doesn't seem right, you can use the hook directly without reassigning it to another variable
```suggestion
  const { page } = useRuntime()
```

File: react/OrderForm.tsx [hunk 2]
```diff
@@ -77,11 +80,17 @@ const getLocalOrderForm = (): OrderForm | null => {
...
+
   const { loading, data, error } = useQuery<{
     orderForm: OrderForm
   }>(OrderFormQuery, {
     ssr: false,
     fetchPolicy: 'no-cache',
+    variables: {
+      refreshOutdatedData: page.includes(CHECKOUT)
```
Comment (lucasecdb) [inline react/OrderForm.tsx:92]: have you checked that if you navigate from, say, a product page to the cart page this query will be remade? or do we need to add an `useEffect` and manually call `refetch`?

File: react/OrderForm.tsx (hunk 1 above)
Comment (silviasfon) [reply react/OrderForm.tsx:84]: I tried doing this but I get the following message
> Property 'page' does not exist on type '() => PublicRuntime'.

File: react/OrderForm.tsx (hunk 2 above)
Comment (silviasfon) [reply react/OrderForm.tsx:92]: so far in all the tests done it is being called

File: react/OrderForm.tsx (hunk 1 above)
Comment (wdsrocha) [reply react/OrderForm.tsx:84]: Did you run `vtex setup --typings`?

Comment (wdsrocha) [reply react/OrderForm.tsx:84]: Btw I think some things will crash after doing this :/, but check if it solves at least for this part of code, pls

Comment (silviasfon) [reply react/OrderForm.tsx:84]: it still doesn't work, I think its because you need to do `runtime().page` instead of `runtime.page`

Comment (lucasecdb) [reply react/OrderForm.tsx:84]: I didn't see the type error you commented above @silviasfon, can you post a screenshot?

**Comment by wdsrocha** [comment]
LGTM at a functional level, but something that I never saw before happened when I got to the cart page:

image

I don't think it's related but just wanted to share 🤔

File: react/OrderForm.tsx (hunk 1 above)
Comment (silviasfon) [reply react/OrderForm.tsx]: It's fixed hahaha

Comment (lucasecdb) [reply react/OrderForm.tsx]: nice!

**Review by lucasecdb — APPROVED** [review]

**Review by wdsrocha — APPROVED** [review]
🔥

**Pull request merged**